packages/nextjs/src/config/loaders/ast.ts

```diff
@@ -0,0 +1,219 @@
+    // stop filtering here. If this ever becomes a problem, more precise filter checks can be found in a comment at the
+    // bottom of this file.
+    return path.node.name === name;
+  };
+
+  return ast.find(Identifier).filter(identifierFilter);
+}
+
+/**
+ * Find all nodes which are declarations of variables with the given name
+ *
+ * @param ast The code, in AST form
+ * @param name The variable name to search for
+ * @returns A collection of NodePaths pointing to any nodes which were found
+ */
+export function findDeclarations(ast: AST, name: string): AST<VariableDeclarationNode> {
+  // Check for a structure of the form
+  //
+  //     node: VariableDeclaration
+  //      \
+  //       declarations: VariableDeclarator[]
+  //        \
+  //         0 : VariableDeclarator
+  //          \
+  //           id: Identifier
+  //            \
+  //             name: string
+  //
+  // where `name` matches the given name.
+  const declarationFilter = function (path: ASTPath<VariableDeclarationNode>): boolean {
+    return (
+      path.node.declarations.length === 1 &&
+      VariableDeclarator.check(path.node.declarations[0]) &&
+      Identifier.check(path.node.declarations[0].id) &&
+      path.node.declarations[0].id.name === name
+    );
+  };
+
+  return ast.find(VariableDeclaration).filter(declarationFilter);
+}
+
+/**
+ * Find all nodes which are exports of variables with the given name
+ *
+ * @param ast The code, in AST form
+ * @param name The variable name to search for
+ * @returns A collection of NodePaths pointing to any nodes which were found
+ */
+export function findExports(ast: AST, name: string): AST<ExportSpecifierNode> {
+  const exportFilter = function (path: ASTPath<ExportSpecifierNode>): boolean {
+    return ExportSpecifier.check(path.node) && path.node.exported.name === name;
+  };
+
+  return ast.find(ExportSpecifier).filter(exportFilter);
+}
+
+/**
+ * Remove comments from all nodes in the given AST.
+ *
+ * Note: Comments are not nodes in and of themselves, but are instead attached to the nodes above and below them.
+ *
+ * @param ast The code, in AST form
+ */
+export function removeComments(ast: AST): void {
+  const nodesWithComments = ast.find(Node).filter(path => !!path.node.comments);
+  nodesWithComments.forEach(path => (path.node.comments = null));
+}
+
+/**
+ * Find an unused identifier name in the AST by repeatedly adding underscores to the beginning of the given original
+ * name until we find one which hasn't already been taken.
+ *
+ * @param userAST The AST to search
+ * @param origName The original name we want to alias
+ * @returns
+ */
+export function findAvailibleAlias(userAST: AST, origName: string): string {
+  let foundAvailableName = false;
+  let newName = origName;
+
+  while (!foundAvailableName) {
+    // Prefix the original function name (or the last name we tried) with an underscore and search for identifiers with
+    // the new name in the AST
+    newName = `_${newName}`;
```

Comment:
Can we prefix with `_sentry_` just to make sure it's easier for users to recognize what's happening if they have to debug this problem?

Reply:
Luca suggested that, too, but I feel like that's misleading, because it's still their version, not the sentrified version. If they do have to debug it, I feel like the `export const getServerSideProps = SentrySDK.withSentryGSSP(_getServerSideProps)` will be a fairly good indication that it's us, no?

Reply:
Fair, the wrapping should be a good indication. Is there a way it’ll fail after we do replacement but before we inject the wrapping `SentrySDK.withSentryGSSP` wrapping?